### PR TITLE
Ender3v3se pin updates

### PR DESCRIPTION
**Pull request by danwel3**
### Description

Updates to pins for the Creality Ender 3 V3 SE.

### Requirements

- Creality Ender 3 V3 SE w/CR4NS200320C13

### Benefits

- Update for some pin definitions (BLTouch, LCD Screen, EEPROM)
- Add assignment for LCD_SERIAL_PORT. Note that the screen will not work with Marlin at this stage as the screen is using a different instruction set to the Ender 2 DWIN screens.

### CODE REVIEW

**Comment by thinkyhead** [comment]
It looks like so far we don't have any example configs using `BOARD_CREALITY_CR4NS`, so we should put something together for that to get the ball rolling, even if it's a bit broken  — because nothing gets stuff fixed quicker than putting it out broken.

**Comment by ThomasToka** [comment]
@thinkyhead have seen this release?

https://github.com/CrealityOfficial/Ender-3V3-SE/blob/main/Marlin/src/pins/stm32f1/pins_CREALITY_V3_GD303.h

File: Marlin/src/pins/stm32f1/pins_CREALITY_V4.h [hunk 1]
```diff
@@ -193,14 +197,30 @@
...
+  #endif
+  #ifndef EXP3_06_PIN
+    #define EXP3_06_PIN                     PB13
+  #endif
+  #ifndef EXP3_07_PIN
+    #define EXP3_07_PIN                     PB12
+  #endif
+  #ifndef EXP3_08_PIN
+    #define EXP3_08_PIN                     PB15
+  #endif
```
Comment (classicrocker883) [inline Marlin/src/pins/stm32f1/pins_CREALITY_V4.h:223]: its not necessary to have this change

`EXP3_01_PIN` and others get defined here.

Comment (thinkyhead) [reply Marlin/src/pins/stm32f1/pins_CREALITY_V4.h:223]: See the bottom of `pins_CREALITY_CR4NS.h` which defines these pins before including `pins_CREALITY_V4.h`.

**Comment by thinkyhead** [comment]
Thanks @ThomasToka — We can use that branch as reference for the configuration of the Ender-3 V3 SE to some extent. If there are customizations to the code that we have not brought over, well, we can add that Marlin fork to the pile of Marlin forks that we are gradually adapting to the current upstream codebase. I almost have CR-6 SE and Ender-5 S1 working too.

File: Marlin/src/pins/stm32f1/pins_CREALITY_CR4NS.h [hunk 2]
```diff
@@ -76,31 +76,34 @@
...
-//#endif
-
 #if HAS_TMC_UART
   // Reduce baud rate to improve software serial reliability
   #define TMC_BAUD_RATE 19200
 
   // Software serial
   #define X_SERIAL_TX_PIN                   PB12
   #define X_SERIAL_RX_PIN        X_SERIAL_TX_PIN
+  #define X_DIAG_PIN                        PB10
```
Comment (thinkyhead) [inline Marlin/src/pins/stm32f1/pins_CREALITY_CR4NS.h:90]: Dedicated DIAG pins are interesting. Is this board pre-built with Trinamic drivers with UART, or does it have other types of stepper drivers? A board that has dedicated DIAG pins may also make these pins available for Trinamic SPI and standalone drivers.

**Comment by queeup** [comment]
Now we need to solve TJC display issue. Then we can use Ender 3 V3 SE with upstream Marlin.

**Comment by mansona** [comment]
@queeup is there an issue or PR that is keeping track of that display issue? I'm just asking so I can subscribe and follow along 😄

**Comment by queeup** [comment]
@mansona I was spamming. Sorry. This PR is not about TJC display.

**Comment by queeup** [comment]
@mansona, https://github.com/MarlinFirmware/Configurations/pull/1105

**Pull request merged**